test/ngAnimate/animateCssSpec.js

```diff
@@ -634,6 +634,86 @@ describe("ngAnimate $animateCss", function() {
+          inject(function($animateCss, $rootElement) {
+
+          ss.addRule('.action', 'transition: color 1s cubic-bezier(0.25, 0.1, 0.25, 1) 5s;');
+          element.addClass('action');
+
+          var options = {
+            to: { background: 'blue' }
+          };
+
+          var animator = $animateCss(element, options);
+          animator.start();
+          triggerAnimationStartFrame();
+
+          expect(element.css('transition-duration')).toMatch('1s');
+          expect(element.css('transition-delay')).toMatch('5s');
+          expect(element.css('transition-property')).toMatch('color');
+          expect(element.css('transition-timing-function')).toBe('cubic-bezier(0.25, 0.1, 0.25, 1)');
+        }));
+
+        it("should apply all explicit transition properties that are already on the element",
+          inject(function($animateCss, $rootElement) {
+
+          ss.addRule('.action', 'transition-duration: 1s;' +
+                                'transition-timing-function: cubic-bezier(0.25, 0.1, 0.25, 1);' +
+                                'transition-property: color;' +
+                                'transition-delay: 5s');
+          element.addClass('action');
+
+          var options = {
+            to: { background: 'blue' }
+          };
+
+          var animator = $animateCss(element, options);
+          animator.start();
+          triggerAnimationStartFrame();
+
+          expect(element.css('transition-duration')).toMatch('1s');
+          expect(element.css('transition-delay')).toMatch('5s');
+          expect(element.css('transition-property')).toMatch('color');
+          expect(element.css('transition-timing-function')).toBe('cubic-bezier(0.25, 0.1, 0.25, 1)');
+        }));
+
+        it("should use the default transition-property (spec: all) if none is supplied in shorthand",
+          inject(function($animateCss, $rootElement) {
+
+          ss.addRule('.action', 'transition: 1s ease');
+          element.addClass('action');
+
+          var options = {
+            to: { background: 'blue' }
+          };
+
+          var animator = $animateCss(element, options);
+          animator.start();
+          triggerAnimationStartFrame();
+
+          expect(element.css('transition-property')).toBe('all');
+        }));
+
+        it("should use the default easing (spec: ease) if none is supplied in shorthand",
+          inject(function($animateCss, $rootElement) {
+
+          ss.addRule('.action', 'transition: color 1s');
+          element.addClass('action');
+
+          var options = {
+            to: { background: 'blue' }
+          };
+
+          var animator = $animateCss(element, options);
+          animator.start();
+          triggerAnimationStartFrame();
+
+          // IE reports ease in cubic-bezier form
+          expect(element.css('transition-timing-function')).toBeOneOf('ease', 'cubic-bezier(0.25, 0.1, 0.25, 1)');
```

Comment:
Nitpick: Since the `cubic-bezier` equivalents of `ease`/`ease-in`/`linear` are used is several tests (and it's not clear why without a comment, nor is it practical to replicate the comment everywhere), it would be nice to have a constant enumeration of the used values at the top and it throughout the tests (e.g. something like `CUBIC_BEZIER_EQUIVALENT.ease`).